src/code/z_actor.c

```diff
@@ -2527,7 +2537,10 @@ void Actor_FreezeAllEnemies(PlayState* play, ActorContext* actorCtx, s32 duratio
     }
 }
 
-void func_80031B14(PlayState* play, ActorContext* actorCtx) {
+/**
+ * Kill actors on room change and update flags accordingly
+ */
+void Actor_KillAllFromUnloadedRooms(PlayState* play, ActorContext* actorCtx) {
```

Comment:
i think the other stuff its doing beyond just killing actors makes it a bit more broader scope. colliders are cleared, temp flags are reset, and some message context variable is zerod (even tho its unused)

Id say something more like `Actor_ResetForRoomChange` or similar wording can be used. not settled on that exact wording, but on that track

Reply:
yeah when I suggested this name I missed that it did other things
I don't have name ideas rn

Reply:
I think `CollisionCheck_ClearContext` is only there to prevent issues with possibly holding references to actors that were just *deleted* (note this function actually doesn't just kill, and it's very similar to `Actor_UpdateAll`'s actor garbage collection
https://github.com/zeldaret/oot/blob/b57f2162ee71b23643fd4e4133f7455905c2969a/src/code/z_actor.c#L2099-L2104
)

So in hindsight this really isn't the best name, maybe "Delete" would be better than "Kill" for starters

Other than that I think the only thing to worry about is
```c
    actorCtx->flags.tempClear = 0;
    actorCtx->flags.tempSwch &= 0xFFFFFF;
```
(let's ignore the unused message context thing)

Reply:
`_GarbageCollectFromUnloadedRooms`, `_DestroyFromUnloadedRooms`, `_DeleteFromUnloadedRooms` ?

Reply:
I can't think of this function as any less than "notify the actor system a room has been unloaded"

The flags clearing
```c
    actorCtx->flags.tempClear = 0;
    actorCtx->flags.tempSwch &= 0xFFFFFF;
```
is pretty important,
clearing `tempClear` means enemies in a cleared room respawn unless something (like the chest actor) set the permanent clear flag,
and clearing switch flags 0x30..0x3F is useful for some puzzles resetting when reentering the room

So I agree with Fig on naming this along the lines of "process stuff when room changes/unloads"

... `Actor_ProcessRoomUnload` ? <img src="https://cdn.discordapp.com/emojis/628286334066491412.webp?size=240&quality=lossless" alt="4Head" height="24">

Reply:
id prob be ok with ProcessRoomChange, though I think including clear or reset in there somehow is appropriate (like my original suggestion, `Actor_ResetForRoomChange`)

i dont think we need to say unload, since the actor system isnt really aware of any loading or unloading thats going on. just that the room is changing (hence the check on cur and prev room)

Reply:
I feel like "room changing" may happen at the start of a room change though, while "room unload" sounds more like it happens at the end

Reply:
its the actor system handling its own end of the entire room change process

Reply:
I mean "room change" is a few parts, including "load next room" "both rooms are loaded, wait one of them unload" "unload other room"

Reply:
right, and my point is that the actor system isnt concerned with those details about whats loaded when. it is simply handling its own things for the room change process as a whole. whether the room is loaded in memory or not isnt really relevant

Reply:
Yeah the room being in memory isn't super relevant, but I think being at the end of the room change is, it would be weird to kill prev rooms' actors at the *start* of the transition
For example with EnHoll, loading another room would trigger a "room change", but you wouldn't want that to instantly delete all actors, in particular if the player went back into the room it would just mean the room got depopulated for no reason

Reply:
well, I dont think unload is sufficent to describe that aspect that youre looking for then. Its hinging on things being loaded or unloaded which is a lower level operation, which is not really relevant to this at all

I still think its fine to describe this as "the function that handles the actor systems tasks for the overall room change process", but I dont think we will agree on that front, and should get some other opinions

Reply:
Well we can always name this however and revisit later once room loading is documented (I want to do that once all transition actors PRs are in)
My impression is `func_80097534` which calls the function we're talking about is something like "commit to loading that next room", maybe having a better view of how the whole system works will bring better terminology

Reply:
agree. id even say its best not to name this right now until rooms have better docs

Reply:
Waiting for @AngheloAlf 's input on this
(would you be fine to not name this right now?)